diff -burN baselayout-2.14.old/Makefile baselayout-2.14/Makefile
--- baselayout-2.14.old/Makefile	2024-04-02 01:42:20.275500539 +0300
+++ baselayout-2.14/Makefile	2024-04-02 01:42:53.816938018 +0300
@@ -52,7 +52,7 @@
 install:
 	$(INSTALL_DIR) $(DESTDIR)/etc
 	cp -pPR etc/* $(DESTDIR)/etc/
-	echo "Gentoo Base System release ${PV}" > ${DESTDIR}/etc/gentoo-release
+	echo "Argent Linux Base System release ${PV}" > ${DESTDIR}/etc/gentoo-release
 	$(INSTALL_DIR) $(DESTDIR)/lib
 	cp -pPR lib/* $(DESTDIR)/lib/
 	$(INSTALL_DIR) $(DESTDIR)/usr/lib
diff -burN baselayout-2.14.old/make_os_release baselayout-2.14/make_os_release
--- baselayout-2.14.old/make_os_release	2024-04-02 01:06:06.885949722 +0300
+++ baselayout-2.14/make_os_release	2024-04-02 01:08:38.149413031 +0300
@@ -4,12 +4,12 @@
 	exit 1
 fi
 cat <<EOF
-NAME=Gentoo
-ID=gentoo
-PRETTY_NAME="Gentoo Linux"
+NAME=Argent
+ID=argent
+PRETTY_NAME="Argent Linux"
 ANSI_COLOR="1;32"
-HOME_URL="https://www.gentoo.org/"
-SUPPORT_URL="https://www.gentoo.org/support/"
-BUG_REPORT_URL="https://bugs.gentoo.org/"
+HOME_URL="https://www.rogentos.ro/"
+SUPPORT_URL="https://rogentos.ro"
+BUG_REPORT_URL="https://www.gitlab.com/rogentos/"
 VERSION_ID="${1}"
 EOF
