--- baselayout.old/make_os_release	2024-03-12 03:37:07.000000000 +0200
+++ baselayout/make_os_release	2024-10-07 01:06:33.431604151 +0300
@@ -4,12 +4,12 @@
 	exit 1
 fi
 cat <<EOF
-NAME=Gentoo
-ID=gentoo
-PRETTY_NAME="Gentoo Linux"
+NAME=Argent
+ID=argent
+PRETTY_NAME="Argent Linux"
 ANSI_COLOR="1;32"
-HOME_URL="https://www.gentoo.org/"
-SUPPORT_URL="https://www.gentoo.org/support/"
-BUG_REPORT_URL="https://bugs.gentoo.org/"
+HOME_URL="https://argentlinux.io/"
+SUPPORT_URL="https://argentlinux.io/support"
+BUG_REPORT_URL="https://gitlab.com/argent/"
 VERSION_ID="${1}"
 EOF
