diff -Nur pycharm-2018.2.2.old/bin/pycharm.sh pycharm-2018.2.2/bin/pycharm.sh
--- pycharm-2018.2.2.old/bin/pycharm.sh	2018-08-21 12:30:59.000000000 +0300
+++ pycharm-2018.2.2/bin/pycharm.sh	2018-09-04 21:39:22.891312237 +0300
@@ -133,6 +133,15 @@
 fi
 
 JAVA_BIN="$JDK/bin/java"
+
+if [ -z "$JAVA_HOME" ] ; then
+  if [ -r /etc/gentoo-release ] ; then
+    JAVA_HOME=`java-config --jre-home`
+    JDK=`java-config --jre-home`
+    JAVA_BIN="${JDK}/bin/java"
+  fi
+fi
+
 if [ -z "$JDK" -o ! -x "$JAVA_BIN" ]; then
   message "No JDK found. Please validate either PYCHARM_JDK, JDK_HOME or JAVA_HOME environment variable points to valid JDK installation."
   exit 1
