diff -Nur a/branding.desc b/branding.desc
--- a/branding.desc	2016-03-06 10:28:28.000000000 +0000
+++ b/branding.desc	2016-08-21 19:06:34.280155216 +0100
@@ -10,9 +10,9 @@
     shortVersionedName:  Argent 1.0
     bootloaderEntryName: Argent
     productUrl:          http://rogentos.ro/
-    supportUrl:          http://rogentos.ro/#contact/
-    knownIssuesUrl:      http://rogentos.ro/#about/
-    releaseNotesUrl:     http://rogentos.ro/#about/
+    supportUrl:          https://discord.gg/hpmt5KU4yT
+    knownIssuesUrl:      https://github.com/rogentos/argent-ws/issues
+    releaseNotesUrl:     https://argentlinux.io
 
 images:
     productLogo:         "argent.png"
diff -Nur a/show.qml b/show.qml
--- a/show.qml	2016-03-06 10:29:56.000000000 +0000
+++ b/show.qml	2016-08-21 19:06:55.288153721 +0100
@@ -24,7 +24,7 @@
     id: presentation
 
     Timer {
-        interval: 5000
+        interval: 50000
         running: true
         repeat: true
         onTriggered: presentation.goToNextSlide()
