diff -Nur WebStorm-211.7142.46.old/bin/webstorm.sh WebStorm-211.7142.46/bin/webstorm.sh
--- WebStorm-211.7142.46.old/bin/webstorm.sh	2021-05-18 07:52:48.139766630 +0300
+++ WebStorm-211.7142.46/bin/webstorm.sh	2021-05-18 07:53:25.419294294 +0300
@@ -79,6 +79,15 @@
   fi
 fi
 
+if [ -z "$JAVA_HOME" ] ; then
+  if [ -r /etc/gentoo-release ] ; then
+    JAVA_HOME=`java-config --jre-home`
+    JDK=`java-config --jre-home`
+    JRE="${JDK}"
+    JAVA_BIN="${JDK}/bin/java"
+  fi
+fi
+
 if [ -z "$JRE" ]; then
   JAVA_BIN=$(command -v java)
 else
