diff -Nur clion-2021.1.1.old/bin/clion.sh clion-2021.1.1/bin/clion.sh
--- clion-2021.1.1.old/bin/clion.sh	2021-05-18 08:56:58.043106255 +0300
+++ clion-2021.1.1/bin/clion.sh	2021-05-18 08:57:37.581596119 +0300
@@ -79,6 +79,15 @@
   fi
 fi
 
+if [ -z "$JAVA_HOME" ] ; then
+  if [ -r /etc/gentoo-release ] ; then
+    JAVA_HOME=`java-config --jre-home`
+    JDK=`java-config --jre-home`
+    JRE="${JDK}"
+    JAVA_BIN="${JDK}/bin/java"
+  fi
+fi
+
 if [ -z "$JRE" ]; then
   JAVA_BIN=$(command -v java)
 else
