diff -Nuar klibc-2.0.8.orig/scripts/Kbuild.klibc klibc-2.0.8/scripts/Kbuild.klibc
--- klibc-2.0.8.orig/scripts/Kbuild.klibc	2021-05-13 15:35:05.603848136 -0000
+++ klibc-2.0.8/scripts/Kbuild.klibc	2021-05-13 15:44:13.163161267 -0000
@@ -81,7 +81,10 @@
 KLIBCCFLAGS       :=
 
 # Defaults for arch to override
-KLIBCARCHINCFLAGS =
+KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/$(KLIBCARCH)/include/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/$(KLIBCARCH)/include/generated/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/$(KLIBCARCH)/include/generated \
+                    -I$(KLIBCKERNELOBJ)/arch/$(KLIBCARCH)/include
 
 # Arch specific definitions for klibc
 include $(srctree)/$(KLIBCSRC)/arch/$(KLIBCARCHDIR)/MCONFIG
@@ -120,8 +123,11 @@
 
 # kernel include paths
 KLIBCKERNELSRC	 ?= $(srctree)
-KLIBCCPPFLAGS    += -I$(KLIBCKERNELSRC)/include		\
-                     $(if $(KBUILD_SRC),-I$(srctree)/include)  \
+KLIBCCPPFLAGS    += -I$(KLIBCKERNELSRC)/include/uapi            \
+                    -I$(KLIBCKERNELSRC)/include/generated/uapi  \
+                    -I$(KLIBCKERNELSRC)/include/generated       \
+                    -I$(KLIBCKERNELSRC)/include                 \
+		     $(if $(KBUILD_SRC),-I$(srctree)/include)   \
 		     $(KLIBCARCHINCFLAGS)
 
 # compiler/assembler option for whether we want an executable stack
diff -Nuar klibc-2.0.8.orig/usr/klibc/arch/i386/MCONFIG klibc-2.0.8/usr/klibc/arch/i386/MCONFIG
--- klibc-2.0.8.orig/usr/klibc/arch/i386/MCONFIG	2021-05-13 15:35:05.594848263 -0000
+++ klibc-2.0.8/usr/klibc/arch/i386/MCONFIG	2021-05-13 15:46:52.989917551 -0000
@@ -10,6 +10,11 @@
 # Enable this to compile with register parameters; only safe for
 # gcc >= 3
 
+KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/x86/include/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/x86/include/generated/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/x86/include/generated \
+                    -I$(KLIBCKERNELOBJ)/arch/x86/include
+
 ifeq ($(CONFIG_REGPARM),y)
 REGPARM_OPT := -mregparm=3 -D_REGPARM=3
 endif
diff -Nuar klibc-2.0.8.orig/usr/klibc/arch/ppc/MCONFIG klibc-2.0.8/usr/klibc/arch/ppc/MCONFIG
--- klibc-2.0.8.orig/usr/klibc/arch/ppc/MCONFIG	2021-05-13 15:35:05.592848291 -0000
+++ klibc-2.0.8/usr/klibc/arch/ppc/MCONFIG	2021-05-13 15:47:39.156269448 -0000
@@ -7,6 +7,11 @@
 # accordingly.
 #
 
+KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/powerpc/include/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/powerpc/include/generated/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/powerpc/include/generated \
+                    -I$(KLIBCKERNELOBJ)/arch/powerpc/include
+
 gcc_m32_option  := $(call cc-option, -m32, )
 
 KLIBCOPTFLAGS	   += -Os
diff -Nuar klibc-2.0.8.orig/usr/klibc/arch/ppc64/MCONFIG klibc-2.0.8/usr/klibc/arch/ppc64/MCONFIG
--- klibc-2.0.8.orig/usr/klibc/arch/ppc64/MCONFIG	2021-05-13 15:35:05.592848291 -0000
+++ klibc-2.0.8/usr/klibc/arch/ppc64/MCONFIG	2021-05-13 15:48:23.243650531 -0000
@@ -7,6 +7,11 @@
 # accordingly.
 #
 
+KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/powerpc/include/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/powerpc/include/generated/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/powerpc/include/generated \
+                    -I$(KLIBCKERNELOBJ)/arch/powerpc/include
+
 KLIBCARCHREQFLAGS = -m64
 KLIBCARCHREQFLAGS += $(call cc-option, -mcall-aixdesc, )
 KLIBCARCHREQFLAGS += $(call cc-option, -mcmodel=small, )
diff -Nuar klibc-2.0.8.orig/usr/klibc/arch/x86_64/MCONFIG klibc-2.0.8/usr/klibc/arch/x86_64/MCONFIG
--- klibc-2.0.8.orig/usr/klibc/arch/x86_64/MCONFIG	2021-05-13 15:35:05.591848305 -0000
+++ klibc-2.0.8/usr/klibc/arch/x86_64/MCONFIG	2021-05-13 15:49:07.617027598 -0000
@@ -14,6 +14,12 @@
 # binaries (20% smaller), but makes the code completely useless for
 # debugging using gdb.
 #
+
+KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/x86/include/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/x86/include/generated/uapi \
+                    -I$(KLIBCKERNELOBJ)/arch/x86/include/generated \
+                    -I$(KLIBCKERNELOBJ)/arch/x86/include
+
 KLIBCARCHREQFLAGS = -m64
 KLIBCOPTFLAGS     += -Os -fomit-frame-pointer -mno-sse \
                      $(call cc-option,-falign-functions=1, )	\
